docs/src/DeveloperDocumentation/CachedFunctions.md

```diff
@@ -0,0 +1,323 @@
+# Functions Which Cache Things
+
+Many functions take in a keyword argument `cached::Bool`, which then cache or do not cache the results of computations to make things faster. This can sometimes lead to strange behaviour. So we document all the functions which do this, and at a future date, maybe do something with this list. 
```

Comment:
"cache or do not cache the results of computations to make things faster"
I don't think this is what is happening here.

To my understanding there are two reasons for caching in OSCAR:
* Functions which do complex computations and cache the result (like `groebner_basis` or `maximal_order`), so it does not need to be computed again. These functions usually do not have (maybe never have) a `cached` keyword.
* Constructors which add the constructed parent object to a global dictionary (= cache), so that if people construct the same polynomial ring or whatever twice, they get the identical ring and everything will *just work*. This is what the `cached` keyword is for. (So it is not about speed but convenience.)

Please correct me, if I am wrong here.

Reply:
You might be completely right. For now, I've removed that, and just added a "FIXME". Someone more qualified than me will have to figure out how to best phrase that paragraph.